kafka/changestream/filter.go

```diff
@@ -0,0 +1,79 @@
+package changestream
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	kf "github.com/codeuniversity/smag-mvp/kafka"
+	"github.com/codeuniversity/smag-mvp/worker"
+
+	"github.com/segmentio/kafka-go"
+)
+
+// Filter is responsible for reading a the change stream,
+// filtering out the events that are not interesting to us
+// and writing new messages based on the changes to the filtered topic
+type Filter struct {
+	*worker.Worker
+
+	changesReader  *kafka.Reader
+	filteredWriter *kafka.Writer
+
+	filterFunc FilterFunc
+}
+
+// FilterFunc given a ChangeMessage from the changesTopic
+// returns zero, one or multiple kafka Messages that should be written to the filteredTopic
+type FilterFunc func(*ChangeMessage) ([]kafka.Message, error)
```

Comment:
Why do we actually want to return s slice of messages?
Isn't one `ChangeMessage` always one change to the database aka. one `kafkaMessage`?

Reply:
One `ChangeMessage` / one changed row in the database can mean multiple messages we want to write to the output topic; i.e. when we have multiple URLs we want to download for one post, it is a bit cleaner to write one message per "download job" instead of one message representing one big job